File: C++/A/716A-crazycomputer.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, c;
    cin >> n >> c;
    vector<int> speeds(n);
    for (int i = 0; i < n; i++)
        cin >> speeds[i];
    int wordcount = 1;
    for (int i = 1; i < n; i++)
        (speeds[i] - speeds[i - 1] <= c) ? wordcount++ : wordcount = 1;
    cout << wordcount;
}

File: Python/A/716A-crazycomputer.py
n, c = map(int, input().split())
speeds = list(map(int, input().split()))
wordcount = 1

for i in range(1, n):
    flag = (speeds[i] - speeds[i - 1] <= c)
    wordcount = wordcount + 1 if flag else 1
print(wordcount)
